### PR TITLE
fixes #8514, TypeApi#baseType exposed into Tasty

### LINKED ISSUE #8514: TASTY: no way to access `baseClasses` of `AppliedType`
I'm working on re-implementation of izumi-reflect for Dotty.

So far I made some substantial progress but seems like there is a blocker.

The primary problem is: seems like there is no way to get access to `baseClasses` of a type:

Let's assume that we have the following definitions:

```scala
trait X[A, B]
trait A[F[_]]
trait B extends A[X[Int, *]]
```

On scala2 I can call `.baseClasses` method on a type of `B` and get a list containing `A[[T] => X[Int, T]]`.

In dotty `AppliedType` has only `tycon` method which does not allow me to inspect applied parents. 

See also: https://github.com/7mind/dotty-typetag-research/issues/2

### PR DESCRIPTION
**Pull request by pshirshov**

### CODE REVIEW

**Comment by smarter** [comment]
Please sign the Scala CLA: https://www.lightbend.com/contribute/cla/scala

**Comment by pshirshov** [comment]
@nicolasstucki : I kinda can't understand how do I run these tests in `run-macros`, so I didn't add a test.

**Comment by smarter** [comment]
Macro tests are bootstrapped only, so you can run them using `dotty-bootstrapped/testCompilation tests/run-macros/foo` (and they're also run by the CI in the test_bootstrapped job)

**Comment by pshirshov** [comment]
@smarter : thanks, that worked. Pushed a basic test. Seems like that should be enough to finish `izumi-reflect`.

File: tests/run-macros/i8514b/Macro_1.scala [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+inline def test(): Unit = ${ testExpr }
+
+def testExpr(using QuoteContext): Expr[Unit] = {
+  import qctx.tasty._
+
+  val t = '[B].unseal.tpe
+  '{
+    println(${Expr(t.baseClasses.map(b => t.baseType(b).toString))})
+  }
```
Comment (nicolasstucki) [inline tests/run-macros/i8514b/Macro_1.scala:15]: The `.check` file is quite hard to read. Maybe print it this way:

```scala
val baseTypes = t.baseClasses.map(b => t.baseType(b))
val str = Expr(baseTypes.map(_.show).mkString("\n"))
'{ println($str}) }
```

**Comment by pshirshov** [comment]
@nicolasstucki : fixed. Could you accept it now?

File: tests/run-macros/i8514b/Macro_1.scala (hunk 1 above)
Comment (pshirshov) [reply tests/run-macros/i8514b/Macro_1.scala:17]: Done

Comment (nicolasstucki) [reply tests/run-macros/i8514b/Macro_1.scala:17]: Perfect

**Pull request merged**